### ISSUE TITLE
JBang could not find dependency `net.sf.saxon:Saxon-HE:11.4`

### ISSUE THREAD

**Issue by someth2say** [jbangdev/jbang#1652]
**Describe the bug**
JBang 0.109.0 is unable to fetch the dependency `net.sf.saxon:Saxon-HE:11.4`, raising a `ResourceNotFoundException`.

**To Reproduce**
1. Have an script with the following dependencies:
```
///usr/bin/env jbang "$0" "$@" ; exit $?

//JAVA 17+

//DEPS net.sf.saxon:Saxon-HE:11.4

class Hello {

    public static void main(final String... args) { 
        System.out.println("Hello JBang");
     }
}
```

2.  Start a fedora:36 (or fedora:38) container, mounting the script (or making it available inside the container somehow).
```
podman run -ti --entrypoint /bin/sh -v $(pwd):/tmp/coursebook:z registry.fedoraproject.org/fedora:36-x86_64
```
3. Install JBang:
```
> curl -Ls https://sh.jbang.dev | JBANG_DOWNLOAD_VERSION=0.109.0 bash -s - app setup
Downloading JBang 0.109.0...
Installing JBang...
Downloading JDK 17. Be patient, this can take several minutes...
Installing JDK 17...
[jbang] Default JDK set to 17 (17.0.7, current, /root/.jbang/cache/jdks/17)
[jbang] Setting up JBang environment...
Please start a new Shell for changes to take effect
> bash
[root@5a8dd007075a /]# 
```
3. run the script
```
[root@5a8dd007075a /]# cd /tmp/coursebook
[root@5a8dd007075a coursebook]# jbang --verbose run Hello.java
```

5. See error
```
[jbang] [0:193] jbang version 0.109.0
[jbang] [0:203] Resolving resource ref: Hello.java
[jbang] [0:206] Resolved resource ref as: Hello.java (cached as: /tmp/coursebook/Hello.java)
[jbang] [0:224] [ERROR] Could not find net.sf.saxon:Saxon-HE:11.4
dev.jbang.cli.ResourceNotFoundException: Could not find net.sf.saxon:Saxon-HE:11.4
        at dev.jbang.source.resolvers.SiblingResourceResolver.resolve(SiblingResourceResolver.java:55)
        at dev.jbang.source.ProjectBuilder.updateProject(ProjectBuilder.java:512)
        at dev.jbang.source.ProjectBuilder.updateProjectMain(ProjectBuilder.java:468)
        at dev.jbang.source.ProjectBuilder.createSourceProject(ProjectBuilder.java:341)
        at dev.jbang.source.ProjectBuilder.build(ProjectBuilder.java:269)
        at dev.jbang.source.ProjectBuilder.build(ProjectBuilder.java:218)
        at dev.jbang.cli.Run.doCall(Run.java:73)
        at dev.jbang.cli.BaseCommand.call(BaseCommand.java:145)
        at dev.jbang.cli.BaseCommand.call(BaseCommand.java:21)
        at picocli.CommandLine.executeUserObject(CommandLine.java:1953)
        at picocli.CommandLine.access$1300(CommandLine.java:145)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2358)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2352)
        at dev.jbang.cli.JBang$3.handle(JBang.java:148)
        at dev.jbang.cli.JBang$3.handle(JBang.java:143)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)
        at picocli.CommandLine$RunLast.execute(CommandLine.java:2316)
        at picocli.CommandLine.execute(CommandLine.java:2078)
        at dev.jbang.Main.main(Main.java:14)
[jbang] [0:224] If you believe this a bug in jbang, open an issue at https://github.com/jbangdev/jbang/issues
```

**Expected behavior**
Dependencies fetched and script ran succesfully.

**JBang version**
[jbang] [0:194] jbang version 0.109.0
Cache: /home/jordisola/.jbang/cache
Config: /home/jordisola/.jbang
Repository:/home/jordisola/.m2/repository
0.109.0

**Additional context**
JBang 0.108.0 works fine following the same steps.

**Comment by maxandersen** [comment]
that is weird... I can confirm that I can reproduce it.

**Comment by maxandersen** [comment]
@quintesse any idea ? we didn't change anything *except* for the script dep support ...maybe some resolution is now missed or causing different behavior?

https://github.com/jbangdev/jbang/compare/v0.108.0...v0.109.0

**Comment by maxandersen** [comment]
ok; this is weird - the dependency is definitely resolvable.

```
List<String> deps = Arrays.asList("net.sf.saxon:Saxon-HE:11.4");

		ModularClassPath classpath = DependencyUtil.resolveDependencies(deps, Collections.emptyList(), false, false,
				true, false);

```
 
returns 9 artifacts.

**Comment by someth2say** [comment]
I think I found the culprit.
In 0.109.0, you introduced the `dev.jbang.source.TagReader#isGav` method, and you start using it when collecting dependencies.
This method delegates to the regex `dev.jbang.dependencies.DependencyUtil#lenientGavPattern` to validate that the dependency is "correct".
But that regex does only accept lowercase `groupId` and `artifactid` names. Hence `net.sf.saxon:Saxon-HE:11.4`is refuses, as it contains a bunch of uppercase letters.

I created a PR that should fix that: https://github.com/jbangdev/jbang/pull/1655
